cluster/Dockerfile

```diff
@@ -16,17 +16,57 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
         unzip \
         zip \
         zlib1g-dev \
-        python3 \
-        python3-numpy \
-        python3-dev \
-        python3-pip \
-        python3-wheel \
-        python3-setuptools \
+        openjdk-8-jdk \
+        openjdk-8-jre-headless \
+				python3 \
+				python3-numpy \
+				python3-dev \
+				python3-pip \
+				python3-wheel \
+				python3-setuptools \
         && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists/*
 
-WORKDIR /app
+# Running bazel inside a `docker build` command causes trouble, cf:
+#   https://github.com/bazelbuild/bazel/issues/134
+# The easiest solution is to set up a bazelrc file forcing --batch.
+RUN echo "startup --batch" >>/etc/bazel.bazelrc
+# Similarly, we need to workaround sandboxing issues:
+#   https://github.com/bazelbuild/bazel/issues/418
+RUN echo "build --spawn_strategy=standalone --genrule_strategy=standalone" \
+    >>/etc/bazel.bazelrc
+# Install the most recent bazel release.
+ENV BAZEL_VERSION 0.5.4
```

Comment:
this is an ancient version of bazel. Let's use a more recent version. Like 0.10.0
https://github.com/bazelbuild/bazel/releases

Reply:
Ah, this should be Bazel 0.8.0 -- https://www.tensorflow.org/install/install_sources#common_installation_problems

Reply:
Done.